GRAMMAR.md

```diff
@@ -0,0 +1,135 @@
+# Floating Point:
+
+Floating point values are defined by the following syntax diagram:
```

Comment:
Add in something saying that this is the _exact_ same definition used in C, C++, and Java (and probably Python too)